### PR TITLE
PR - Speech to Text and Teams Extension

### PR DESCRIPTION
**Pull request by hunterjam**
## Purpose

* These changes include: speech-to-text via the web interface, a Teams extension, and updated README content.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
Follow the updated steps on the README file.
```

* Test the code

```
Follow the updated steps on the README file.
```

## What to Check
Verify that the following are valid
* When loading the web chat interface, verify the microphone icon is available. Click the icon and verify the microphone is working.

### CODE REVIEW

File: LOCAL_DEPLOYMENT.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
+[Back to *Chat with your data* README](README.md)
```
Comment (ross-p-smith) [inline LOCAL_DEPLOYMENT.md:1]: Should we put this in the docs folder?

File: TEAMS_EXTENSION.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
+[Back to *Chat with your data* README](README.md)
+
```
Comment (ross-p-smith) [inline TEAMS_EXTENSION.md:2]: Should we put this in the docs folder?

**Comment by ross-p-smith** [comment]
This looks awesome I am going to try this out!

File: TEAMS_EXTENSION.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+2. Under **Environment**, select **test**.
+
+![Teams Toolkit Environment](/media/teams-4.png)
+3. Under **Lifecycle**, select **Provision**.
+
+![Teams Toolkit Lifecycle Provision](/media/teams-5.png)
+4. Within the command palette, select **test** for the environment.
+
+![Select an Environment](/media/teams-6.png) 
+5. Select the resource group created earlier in the installation
```
Comment (ross-p-smith) [inline TEAMS_EXTENSION.md:66]: Needs a new line after each image in this numbered list as text is not aligned

File: TEAMS_EXTENSION.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+![Select a Resource Group](/media/teams-7.png) 
+6. When prompted about Azure charges, select **Provision**.
+
+![Azure Charges Prompt](/media/teams-8.png)
+7. Verify that the provisioning was successful.
```
Comment (ross-p-smith) [inline TEAMS_EXTENSION.md:72]: At this stage I get an error that I have not set `APP_NAME_SUFFIX` - This is missing in the `.env.test` file.

File: code/app/app.py [hunk 5]
```diff
@@ -23,7 +23,18 @@
 @app.route("/<path:path>")
 def static_file(path):
     return app.send_static_file(path)
-
+@app.route('/api/config', methods=['GET'])
+
+def get_config():
```
[the remark this answers is not in the document]
Comment (hunterjam) [reply code/app/app.py:28]: The Azure Speech service (via SDK) requires the AZURE_SPEECH_SERVICE_KEY and AZURE_SPEECH_SERVICE_REGION. (Line 159 within the Chat.tsx file). The API is running over a secure connection, correct? Returning this information should be okay. If you have a recommendation for improvement, let me know.

File: LOCAL_DEPLOYMENT.md (hunk 1 above)
Comment (hunterjam) [reply LOCAL_DEPLOYMENT.md]: Good callout. Done.

File: TEAMS_EXTENSION.md (hunk 2 above)
Comment (hunterjam) [reply TEAMS_EXTENSION.md]: Good callout. Done.

File: TEAMS_EXTENSION.md (hunk 3 above)
Comment (hunterjam) [reply TEAMS_EXTENSION.md]: Good find. Done.

File: TEAMS_EXTENSION.md (hunk 4 above)
Comment (hunterjam) [reply TEAMS_EXTENSION.md]: Good find. Done.

**Comment by hunterjam** [comment]
@ross-p-smith --- I added documentation for Teams local deployment.

**Review by adamdougal — COMMENTED** [review]
This looks great!

File: code/app/app.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (hunterjam) [reply code/app/app.py:28]: Completed: https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/issues/101

File: extensions/backend/utilities/common/Answer.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
+import json
```
[the remark this answers is not in the document]
Comment (hunterjam) [reply extensions/backend/utilities/common/Answer.py:1]: This is a discussion I want to have with the larger team. For the Teams extension, I needed to set this up as an Azure Function. Would it make more sense to have the Frontend Web App also use the same Azure Function (for simplification)? If so, can we log this as an issue for the backlog?

[the remark this answers is not in the document]
Comment (hunterjam) [reply extensions/backend/utilities/common/Answer.py]: Updated to copy utilities from "code" folder.

[the remark this answers is not in the document]
Comment (adamdougal) [reply extensions/backend/utilities/common/Answer.py]: Ah great thanks! I was going to say that I was happy to defer and raise an issue but looks like it was fairly straight forward.

**Comment by adamdougal** [comment]
This LGTM! @ross-p-smith Did you get a chance to try this out? Are you happy to merge?

**Review by ross-p-smith — COMMENTED** [review]
Thanks for this @hunterjam

**Comment by hunterjam** [comment]
Just want to add that @blessing-msft also contributed to this PR.

**Comment by gmndrg** [comment]
Thank you so much @hunterjam, @blessing-msft  and team for all the hard work to get this into the accelerator. This is an amazing functionality.

**Pull request merged**